### ISSUE TITLE
*ERROR*: Assertion failed: (eq (match-end 0) (length string))

### ISSUE THREAD

**Issue by caedmon-kitty** [xonsh/xontrib-abbrevs#6]
I've found that ...
With this abbrev - **abbrevs['ec'] = 'emacsclient -c -n'**

I enter ec _filespec_ at the prompt and expect an emacs frame to pop up.  But if the _filespec_ begins with a hidden directory folowed by the filename (**ec .Xincludes/xterm** for example) this error message is printed -

*ERROR*: Assertion failed: (eq (match-end 0) (length string))

This appears to only occur if the directory portion of the _filespec_ begins with a dot and the filename immediately follows the directory name.  If the directory name is followed by another directory then all is ok.

In other words this results in the error -

**ec .thunderbird/profile.ini**

while this works as expected 

**ec .config/autostart/alacritty.desktop**

since the in the latter example the .config directory is followed by the autostart directory then the filename.

xonsh is version 14.1 installed via pip
Python 3.10.12
ubuntu 22.04 LTS

**Comment by caedmon-kitty** [comment]
Forgot to mention - if I do not use the abbrev there is no error - this works without an error message

**emacsclient -c -n .Xincludes/xterm**
